Generator/Gir/GSignal.cs

```diff
@@ -15,7 +15,7 @@ public class GSignal
         public GDoc? Doc { get; set; }
 
         [XmlElement("parameters", Namespace = "http://www.gtk.org/introspection/core/1.0")]
-        public GParameters? Parameters { get; set; }
+        public GParameters Parameters { get; set; } = default!;

```

Comment:
If you write it like this, it means Parameters can not be null, meaning in every XML there is a `Parameters` field for signals. (The flow analysis of roslyn will not check for nulls.)

But eg. for the GTK-Button clicked event there are no parameters:

```XML
      <glib:signal name="clicked" when="first" action="1">
        <doc xml:space="preserve" filename="gtk/gtktoolbutton.c" line="318">This signal is emitted when the tool button is clicked with the mouse
or activated with the keyboard.</doc>
        <return-value transfer-ownership="none">
          <type name="none" c:type="void"/>
        </return-value>
      </glib:signal>
```

For optional Properties like `Parameters` it should be like `public GParameters? Parameters { get; set; }`. If you check inside your code for nulls the warnings should go away.